cmd/kube-controller-manager/app/controllermanager.go

```diff
@@ -484,30 +484,38 @@ func (c serviceAccountTokenControllerStarter) startServiceAccountTokenController
 		glog.Warningf("%q is disabled because there is no private key", saTokenControllerName)
 		return false, nil
 	}
-	privateKey, err := certutil.PrivateKeyFromFile(ctx.ComponentConfig.KubeCloudShared.ServiceAccountKeyFile)
-	if err != nil {
-		return true, fmt.Errorf("error reading key for service account token controller: %v", err)
-	}
 
-	var rootCA []byte
+	var (
+		rootCA []byte
+		err    error
+	)
 	if ctx.ComponentConfig.SAController.RootCAFile != "" {
 		rootCA, err = ioutil.ReadFile(ctx.ComponentConfig.SAController.RootCAFile)
 		if err != nil {
 			return true, fmt.Errorf("error reading root-ca-file at %s: %v", ctx.ComponentConfig.SAController.RootCAFile, err)
 		}
-		if _, err := certutil.ParseCertsPEM(rootCA); err != nil {
+		if _, err = certutil.ParseCertsPEM(rootCA); err != nil {
```

Comment:
revert this, shadowing vars is not preferred

Reply:
@liggitt  Did you mean we should change 
`if _, err = certutil.ParseCertsPEM(rootCA); err != nil { `  to
`if _, err := certutil.ParseCertsPEM(rootCA); err != nil {`
I am confused since I think what I am doing here is avoid shadowing err.
Did I misunderstanding your opinions?

Reply:
Sorry, I misspoke… it's preferable to keep the error local to the block (which happens to be shadowing in this case), rather than worrying about what side effects it has outside the block